components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java

```diff
@@ -706,47 +732,231 @@ public static String getUserConsentURL(OAuth2Parameters params, String loggedInU
         return consentPage;
     }
 
-    private static String getFilteredScopes(OAuth2Parameters params) throws OAuthSystemException {
+    /**
+     * Deny consent for given set of OAuth scopes for the application.
+     *
+     * @param params    OAuth2 request parameters.
+     * @param user      Authenticated user.
+     * @throws OAuthSystemException
+     */
+    public static void handleDeniedOAuthScopeConsent(OAuth2Parameters params, AuthenticatedUser user)
+            throws OAuthSystemException {
 
         try {
-            Set<String> allowedScopes = params.getScopes();
-            startTenantFlow(params.getTenantDomain());
+            Set<String> userDeniedScopesSet = params.getConsentRequiredScopes();
+            if (CollectionUtils.isNotEmpty(userDeniedScopesSet)) {
+                List<String> disapprovedScopeConsents = new ArrayList<>(userDeniedScopesSet);
+                // Remove OIDC scopes.
```

Comment:
Lets add a debug log to see the denied consents

Reply:
Fixed in https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1548/commits/27ddbc8fed9c4368ffe7a826b857537a6b8ffff0